stock_request_direction/models/stock_request_order.py

```diff
@@ -24,6 +24,14 @@ def _onchange_location_id(self):
             self.location_id = \
                 self.warehouse_id.lot_stock_id.id
 
+    @api.onchange('warehouse_id')
+    def _onchange_warehouse_id(self):
+        if self.direction:
+            self.direction = False
```

Comment:
@patrickrwilson If you change the warehouse, the direction should remain the same, no?

Reply:
@max3903 When the direction stays set, it doesn't seem to update the location and route information available on the line items so by clearing that and the routes, then forcing the user to set the direction again it then resets the location properly and also seems to 'reset' what shows for the line items. This is why I cleared the direction.

Reply:
@patrickrwilson what about calling `self._onchange_location_id()`?

Reply:
@max3903 having it call the other on-change doesn't seem to work for me. I have 2 WH, 'Main WH' and 'WH2'. If I have it call the other on-change method instead of making the direction blank, the location changes to the stock location of the warehouse like if it were an inbound. 

Example, I set 'Main WH', direction 'Outbound' and it sets correctly for an outgoing order. Then change the warehouse to 'WH2', now the location changes to 'WH2/Stock' and not the correct one going outbound. If i change the direction to inbound then outbound it resets location correctly. 

This is what i changed the code to:
    @api.onchange('warehouse_id')
    def _onchange_warehouse_id(self):
        if self.direction:
            self._onchange_location_id()
        for stock_request in self.stock_request_ids:
            if stock_request.route_id:
                stock_request.route_id = False

Maybe i'm missing something or my code is wrong. Thanks for your help.